plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy

```diff
@@ -0,0 +1,960 @@
+    private String publisherID
+
+    WrrocRenderer(Map opts) {
+        path = opts.file as Path
+        overwrite = opts.overwrite as Boolean
+
+        ProvHelper.checkFileOverwrite(path, overwrite)
+    }
+
+    private static final List<String> README_FILENAMES = List.of("README.md", "README.txt", "readme.md", "readme.txt", "Readme.md", "Readme.txt", "README")
+
+    @Override
+    void render(Session session, Set<TaskRun> tasks, Map<Path,Path> workflowOutputs) {
+        final params = session.params
+        final configMap = session.config
+
+        // Set RO-Crate Root and workdir
+        this.crateRootDir = Path.of(params['outdir'].toString()).toAbsolutePath()
+        this.workdir = session.getWorkDir()
+        this.projectDir = session.getWorkflowMetadata().getProjectDir()
+
+        // get workflow inputs
+        final taskLookup = ProvHelper.getTaskLookup(tasks)
+        final workflowInputs = ProvHelper.getWorkflowInputs(tasks, taskLookup)
+
+        // Add intermediate input files (produced by workflow tasks and consumed by other tasks)
+        workflowInputs.addAll(getIntermediateInputFiles(tasks, workflowInputs))
+        final workflowInputMapping = getWorkflowInputMapping(workflowInputs)
+
+        // Add intermediate output files (produced by workflow tasks and consumed by other tasks)
+        workflowOutputs.putAll(getIntermediateOutputFiles(tasks, workflowOutputs))
+
+        // Copy workflow input files into RO-Crate
+        workflowInputMapping.each { source, dest ->
+            if( Files.isDirectory(source) ) {
+                // Recursively copy directory and its contents
+                Files.walkFileTree(source, new SimpleFileVisitor<Path>() {
+                    @Override
+                    FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                        Path targetDir = dest.resolve(source.relativize(dir))
+                        Files.createDirectories(targetDir)
+                        return FileVisitResult.CONTINUE
+                    }
+
+                    @Override
+                    FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        Path targetFile = dest.resolve(source.relativize(file))
+                        if (!Files.exists(targetFile))
+                            Files.copy(file, targetFile)
+                        return FileVisitResult.CONTINUE
+                    }
+                })
+            } else {
+                try {
+                    Files.createDirectories(dest.getParent())
+                    if( !Files.exists(dest) )
+                        Files.copy(source, dest)
+                } catch (Exception e) {
+                    println "workflowInput: Failed to copy $source to $dest: ${e.message}"
+                }
+            }
+        }
+
+        // Copy workflow output files into RO-Crate
+        workflowOutputs.each { source, dest ->
```

Comment:
Putting intermediate files aside, the workflow outputs should already be present in the output directory, so there should be no need to copy them. The most I would do is verify that each workflow output actually resides in the output directory and warn the user if it doesn't.